syntax = "proto3";
package voting;

import "google/protobuf/empty.proto";

message VoteRequest {
  string voter_id = 1;
  string candidate_id = 2;
}

message VoteResponse {
  bool success = 1;
  string message = 2;
}

message VoteResults {
  map<string, int32> results = 1; // Ex.: { "CandidatoA": 5, "CandidatoB": 8 }
}

// Mensagem para propagar blocos entre nós
message BlockMessage {
  map<string, string> data = 1;
}

service VotingService {
  // Voto normal, criptografado e gravado localmente + broadcast
  rpc SubmitVote (VoteRequest) returns (VoteResponse);
  // Consulta resultados (descriptografa a própria cadeia)
  rpc QueryResults (google.protobuf.Empty) returns (VoteResults);
  // RPC interno: recebe blocos de outros nós
  rpc PropagateBlock (BlockMessage) returns (VoteResponse);
}
